adoc/admin-monitoring-cluster-health-check.adoc

```diff
@@ -0,0 +1,372 @@
+====
+
+=== {kube} master
+
+All components in {kube} cluster expose a `/healthz` endpoint. The expected (healthy) HTTP response status code is `200`.
+
+The minimal services for the master to work properly are:
+
+- kube-apiserver:
++
+The component that receives your requests from `kubectl` and from the rest of the {kube} components. The URL is https://<MASTER-NODE-IP-ADDRESS/FQDN>:6443/healthz
++
+* Local Check
++
+[source,bash]
+----
+curl -k -i https://localhost:6443/healthz
+----
+* Remote Check
++
+[source,bash]
+----
+curl -k -i https://<MASTER-NODE-IP-ADDRESS/FQDN>:6443/healthz
+----
+
+- kube-controller-manager:
++
+The component that contains the control loop, driving current state to the desired state. The URL is http://<MASTER-NODE-IP-ADDRESS/FQDN>:10252/healthz
++
+* Local Check
++
+[source,bash]
+----
+curl -i http://localhost:10252/healthz
+----
+* Remote Check (make sure firewall allows port `10252`)
++
+[source,bash]
+----
+curl -i http://localhost:10252/healthz
+----
+
+- kube-scheduler:
++
+The component that schedules workloads to nodes. The URL is http://<MASTER-NODE-IP-ADDRESS/FQDN>:10251/healthz
++
+* Local Check
++
+[source,bash]
+----
+curl -i http://localhost:10251/healthz
+----
+* Remote Check (make sure firewall allows port `10251`)
++
+[source,bash]
+----
+curl -i http://localhost:10251/healthz
+----
+
+.High-Availability Environments
+[NOTE]
+====
+In a HA environment you can monitor `kube-apiserver` on `https://MASTER-NODE-LOADBALANCER:6443/healthz`.
+
+If any master node is running correctly you will receive a valid response.
+
+This does, however, not mean that all master nodes necessarily work correctly.
+To ensure that all master nodes work properly, the health checks must be repeated individually for each master node deployed.
+
+This endpoint will return a successful HTTP response if the cluster is operational; otherwise it will fail.
+It will for example check that it can access `etcd` too.
+This should not be used to infer that the overall cluster health is ideal.
+It will return a a successful response even when only minimal operational cluster health exists.
+
+To probe for full cluster health, you must perform individual health checking for all machines individually.
+====
+
+=== `Etcd` Cluster
+
+The etcd cluster expose a endpoint `/health`. The expected (healthy) HTTP response body is `{"health":"true"}`. The etcd cluster is access by HTTPS only. So we need to have etcd certificates.
+
+* Local Check
++
+[source,bash]
+----
+curl --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/healthcheck-client.crt --key /etc/kubernetes/pki/etcd/healthcheck-client.key https://localhost:2379/health
+----
+* Remote Check
++
+The etcd cluster is not allowed remote check due to the etcd server could be access in {kube} clusters only.
+
+== Node Health Checks
+
+The basic check if a node is healthy consists of checking if `kubelet` and the CNI (Container Networking Interface) are working properly.
```

Comment:
This sentence is little bit awkward to me.
maybe) The basic checks of kubelet and CNI in a node indicate that a node is healthy.

Reply:
How about this? *The basic check a node is healthy by checking if `kubelet` and the CNI (Container Networking Interface) are working properly.*

Reply:
Maybe this,
```
Performs basic check of a node health. This consists of checking `kubelet` endpoint and CNI (Container Networking Interface) pod state.
```

Reply:
:100: